core-rust/core-api-server/core-api-schema.yaml

```diff
@@ -7553,6 +7556,32 @@ components:
           $ref: "#/components/schemas/Substate"
         current_time_rounded_to_minutes:
           $ref: "#/components/schemas/Substate"
+        current_validator_readiness_signals:
+          description: |
+            Protocol versions signalled by current Validator set.
+            Only returned if enabled by `include_readiness_signals` on your request.
+          type: array
+          items:
+            $ref: "#/components/schemas/ProtocolVersionReadiness"
+    ProtocolVersionReadiness:
+      type: object
+      required:
+        - signalled_protocol_version
```

Comment:
In terms of our tracking, I think it would actually be helpful to include the validators who have no signalled protocol version, and also the stake proportion of each validator.

(As importantly, in a way, it matters more who _hasn't_ signalled yet than who has)

Reply:
Although thinking about this a little more, I quite like the grouping...

Okay, so conclusion I think - how about we keep the grouping, but:
* We include a grouping with empty signalled protocol version
* We include a `total_active_stake_proportion` on the `ProtocolVersionReadiness` (so across the groups, they should add to something very close to 1)
* We change `signalling_validator_addresses` to `signalling_validators` (of a `SignallingValidator` item) which includes the `validator_address` as well as an `active_stake_proportion` for that validator.

Reply:
hah, I thought about this, and my classic "don't return easily-computable fields" mental barrier kicked in...
(the same response contains a full active validator set, with their stakes, so everything you mentioned will be there)

But ofc we have to answer what is "easily-computable" here:
**A)** if we expect people to call this endpoint manually in order to quickly glance at the readiness field: then I agree to all your suggestions
**B)** if we expect this will be mostly used to render some dashboard: then I say "2 lines of code on the caller's side".

❓ 

_I ack that we are already doing the "convenience" response here (after all, the caller could loop through validators and do 100 more calls themselves) so I won't be surprised if you confirm **A)** above :)_

Reply:
(addressed according to **A)**)